spark/src/main/scala/org/apache/spark/sql/delta/catalog/AbstractDeltaCatalog.scala

```diff
@@ -237,11 +237,7 @@ class AbstractDeltaCatalog extends DelegatingCatalogExtension
     try {
       super.loadTable(ident) match {
         case v1: V1Table if DeltaTableUtils.isDeltaTable(v1.catalogTable) =>
-          DeltaTableV2(
-            spark,
-            new Path(v1.catalogTable.location),
-            catalogTable = Some(v1.catalogTable),
-            tableIdentifier = Some(ident.toString))
+          newDeltaCatalogBasedTable(ident, v1.catalogTable)
```

Comment:
why this change? what does this achieve?

Reply:
I am trying to avoid extending the whole loadTable method, which will have more complex logic like

```
if (isIcebergPathIdentifier(ident)) {
            newIcebergPathTable(ident)
}
```
But change the delta specific logic in the sub class